File: src/main/java/de/huddeldaddel/euler/Problem7.java
package de.huddeldaddel.euler;

import de.huddeldaddel.euler.math.PrimeSequence;
import de.huddeldaddel.euler.math.PrimeSequence;

public class Problem7 {

    public static void main(String[] args) {
        System.out.println(new Problem7().getPrimeNumber(10_001));
    }

    public long getPrimeNumber(final int num) {
        final PrimeSequence primeSequence = new PrimeSequence();
        for(int i=1; i<num; i++)
            primeSequence.nextValue();
        return primeSequence.nextValue();
    }

}

File: src/main/kotlin/de/huddeldaddel/euler/Problem7.kt
package de.huddeldaddel.euler

import de.huddeldaddel.euler.math.PrimeSequence

/**
 * Solution for https://projecteuler.net/problem=7
 */
fun main() {
    println(Problem7().getPrimeNumber(10001))
}

class Problem7 {

    fun getPrimeNumber(num: Int): Long {
        val primeSequence = PrimeSequence()
        for (i in 1 until num)
            primeSequence.nextValue()
        return primeSequence.nextValue()
    }

}
